Sprint 10

tlavr opened the pull request:
Project update according to sprint 10 requirements

tlavr committed:
Fix (review comments) : Refactor progile image code according to sprint 09 pull request comments

tlavr committed:
Feature(Auth): Add AuthView and WebView controllers. Add segue

tlavr committed:
Feature(auth token): Add auth token fetch functions

tlavr committed:
Feature(splash screen): Add splash screen and finish sprint 10 requirements

DanilaG reviewed (commented):
По коду всё классно)

tlavr committed:
Fix (segue type): Fix segue type for SplashScreen -> AuthView from Show to Present modally and full screen

Fix (launch screen logo): Fix image name for launch screen logo

Fix (Login button): Add explicit button outler and IBAction to perform segue towards AuthView

tlavr committed:
Feature (static library?) : Add WebKit, UIKit and Foundation to General project settings

tlavr committed:
Fix (two WebViews) : Removed second WebView opening from the code

tlavr committed:
Fix (progress view color): Change color to YP Black

tlavr commented:
@DanilaG Привет! Можешь, пожалуйста, подсказать с какой ошибкой у тебя падает приложение? У меня просто при тестировании все работает и даже в предыдущей версии работало. Подтягивание и линковка каких-то библиотек кажется не в рамках этого спринта и действую немного в слепую, просто другие ребята подсказывают с чем у них падало... Я у себя не могу воспроизвести такие же ошибки и соответственно вообще не имею представления как дебажить и откуда начинать копать. Такое чувство, что ошибки вызваны разницей в окружении или в библиотеках какой-то?
Записал экран, чтобы показать как у меня это все выглядит: https://disk.yandex.ru/i/ispc0fLj4k5UBw 

Также попытался найти индикатор загрузки в фигме но вроде там его нет в макете? Я поменял с YP Background на YP Black, думаю, это то что ты имел ввиду?

tlavr committed:
Fix(progress view value) : Change starting value to 0.25

tlavr committed:
Fix(WebView dismiss) : Remove second dismiss call in didAuth function

DanilaG commented:
Привет! Сейчас ошибка возникает при первом входе - лента не открывается. Если войти повторно, то открывается. Обсуждаем с Наставником что делать
- https://disk.yandex.ru/i/R5s6R538KcLHlw

Да, цвет действительно в макетах не прописан, так что приму любой

tlavr commented:
@DanilaG Спасибо за видео! То есть я так понял сценарий следующий:
1. Открываешь приложение в первый раз - успешно подгружается экран авторизации
2. Нажимаешь кнопку - Войти - успешно подгружается WebView
3. Вводишь логин и пароль, но вместо перехода на галерею опять открывается экран авторизации? При это на видео видно что галерея промелькнула и опять на экран авторизации выбросило? Но как такого рантам краша не происходит?
И потом при поточном входе я так понимаю токен уже грузится из сохраненных данных и опять же не пускает на галерею?
У меня такое не воспроизводится, если не затруднит можешь скинуть логи или ошибки если такие есть?

tlavr committed:
Feature (profile request): Add profile request and HUD progress view

tlavr committed:
Fix (progress HUD): Add progress HUD

tlavr committed:
Fix (slow internet case): Fix slow internet issue

tlavr committed:
Refactor(token request): Move token request to AuthView and use pop method to dismiss VC

DanilaG commented:
Привет! Прошу прощения, не заметил вопросы. Лучше задавать вопросы через наставника, так их точно не пропущу)

Вводишь логин и пароль, но вместо перехода на галерею опять открывается экран авторизации?
- Сейчас краш возникает не в зависимости от первого или повторного входа. Переход в галерею осуществляется.

При это на видео видно что галерея промелькнула и опять на экран авторизации выбросило?
- Передовая наставнику, что частый кейс, когда подвязывают проверку токена на viewDidAppear. Это может приводить к тому, что web view скроется и сработает viewDidAppear до того как токен запишется в хранилище

Но как такого рантам краша не происходит?
- Сейчас краш происходит в `ImageFeed/SplashViewController.swift:101: Fatal error: Error occured during profile data loading: decoderError`. Наиболее частый кейс такой ошибки - не опциональные поля, которые должны быть опциональными по API (username, firstName, lastName). Обратись к Наставнику, он имеет доступ к "особому аккаунт" для проверки, и сможет тебе проверить/подсказать пофиксилось ли

И потом при поточном входе я так понимаю токен уже грузится из сохраненных данных и опять же не пускает на галерею?
- На сколько я помню, при повторном входе пускало, но сейчас точно пускает)

У меня такое не воспроизводится, если не затруднит можешь скинуть логи или ошибки если такие есть?
- Тут могут быть проблемы как с многопоточкой, так и с особым акаунтом

tlavr committed:
Feature(profile error handling): Add graceful profile errors handling and optional fields processing

tlavr commented:
@DanilaG Привет, поправил обработку опциональных полей профиля и добавил базовый алерт вместо выброса assertionFailure. С наставником обсудили - там возникла небольшая путаница и на самом деле запрос профиля и тд должен был быть в 11 спринте, но решили уже оставить как есть и дальше отрефакторить в 11 спринте под требования как нужно. Вроде бы у наставника с особым акаунтом все прошло и пункты 10 спринта должны быть все реализованы. Также согласовали 4 попытку в рамках 10 спринта из-за этой путаницы :)
